crates/swc_ecma_codegen/src/config.rs

```diff
@@ -5,7 +5,6 @@ use swc_ecma_ast::EsVersion;
 #[derive(Debug, Clone, Copy)]
 #[cfg_attr(feature = "serde-impl", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde-impl", serde(rename_all = "camelCase"))]
-#[non_exhaustive]
```

Comment:
@kdy1 I'm a little confused about the maintenance burden of not having this. It seems like this struct rarely ever changes?

![image](https://github.com/swc-project/swc/assets/1609021/f0f70157-c477-4fad-b154-a82a48453f65)

Of those, it seems like it's changed one time per year, so it just requires a minor version bump on update, especially if all downstream swc crates do not use the struct expression.

Overall, it's not a big deal to have this, I just personally think it's a much better API for consumers to use the struct expression, which is why I opened this PR.

Reply:
I made fewer changes to avoid breaking changes. https://github.com/swc-project/swc/blob/06a2b444993f760fa23d4b4ebc25201c64cde0fa/crates/swc_ecma_codegen/src/text_writer/basic_impl.rs#L49

I think this should go into the `Config` struct, but I didn't want a breaking change.

Reply:
There was many similar situations IIRC

Reply:
Ok, I thought there's a script in swc that automatically updates all the versions on a breaking change. I'll close this PR.

Reply:
Versions in `Cargo.toml` are automatic, but actually the problem is that I don't have control over `mdxjs-rs` and I have to wait for a long time.

Reply:
Oh, yeah I just re-read your other comment.